bip-0360.mediawiki

```diff
@@ -170,386 +172,269 @@ using Grover's algorithm would require at least 10^24 quantum operations. As for
+[https://github.com/bitcoin/bips/blob/master/bip-0173.mediawiki#bech32 BIP-173]. Bech32 encoding maps version 3 to the
+prefix <code>r</code>.
 
-=== Signature Algorithms ===
+Example P2QRH address:
 
-The specific quantum-resistant signature algorithm used cannot be inferred from the length of the public key due to
-collisions in length between algorithms. Instead, when each key is revealed in the attestation, the key type bitmask
-indicates which algorithm was used.
+<code>bc1r...</code> (32-byte Bech32m-encoded tagged hash tapleaf merkle root)
 
-Supported PQC algorithms and their NIST Level I parameters:
+==== ScriptPubKey ====
 
-* '''secp256k1 - BIP-340 - Schnorr + X-Only'''
-** Key Type 0
-** Public Key Length: 32 bytes
-** Signature Length: 64 bytes
-** Total Size: 96 bytes
-** Cycles to sign: 42,000 (EdDSA)
-** Cycles to verify: 130,000 (EdDSA)
-* '''FN-DSA-512 - FIPS 206 - FALCON-512:'''
-** Key Type 1
-** Public Key Length: 897 bytes
-** Signature Length: 667 bytes
-** Total Size: 1,564 bytes
-** Cycles to sign: 1,009,764
-** Cycles to verify: 81,036
-* '''ML-DSA-44 - FIPS 204 - CRYSTALS-Dilithium Level I:'''
-** Key Type 2
-** Public Key Length: 1,312 bytes
-** Signature Length: 2,420 bytes
-** Total Size: 3,732 bytes
-** Cycles to sign: 333,013
-** Cycles to verify: 118,412
-* '''SLH-DSA-SHAKE-128s - FIPS 205 - SPHINCS+-128s:'''
-** Key Type 3
-** Public Key Length: 32 bytes
-** Signature Length: 7,856 bytes
-** Total Size: 7,888 bytes
-** Cycles to sign: 4,682,570,992
-** Cycles to verify: 4,764,084
+The <code>scriptPubKey</code> for a P2QRH output is:
 
-Implementations must recognize the supported algorithms and validate accordingly.
+  OP_PUSHNUM_3 OP_PUSHBYTES_32 <nowiki><hash></nowiki>
 
-A bitmask is used to indicate the algorithm used for each public key and signature pair. The bitmask enumerates based on
-the key type as indicated above. This is used in the cryptographic commitment in the hash computation and
-revealed in the attestation for each public key when spent.
+Where:
 
-=== Script Validation ===
+* <code>OP_PUSHNUM_3</code> (<code>0x03</code>) indicates SegWit version 3.
+* <nowiki><hash></nowiki> is the 32-byte HASH256 of the commitment, as defined in the Hash Commitment section below.
 
-To spend a P2QRH output, the following conditions must be met:
+=== Post-Quantum Signature Algorithms ===
 
-1. The <code>scriptPubKey</code> must be of the form:
+We propose adding the opcodes OP_CHECKSIG_ML (ML-DSA) and OP_CHECKSIG_SLH (SLH-DSA) to tapscript by redefining OP_SUCCESSx
+opcodes. As ML-DSA and SLH-DSA are post-quantum signature algorithms this would allow the user of post-quantum signatures 
+in tapscript.
 
-OP_PUSHNUM_3 <32-byte hash>
+These opcodes function similarly to the existing OP_CHECKSIG opcode. They pop the public key and signature from the stack
+and then perform signature verification using the specified post-quantum signature algorithm.
 
-2. The attestation must include:
+TODO: Open questions:
```

Comment:
Mediawiki renders this and the next two lines as a single paragraph. Either lead all three lines with a space, or remove the ‘TODO: ‘ from the following two lines and use the mediawiki numbered list ‘#’ instead of numbers.